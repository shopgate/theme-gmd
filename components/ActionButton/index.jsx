/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import IndicatorCircle from 'Components/IndicatorCircle';
import RippleButton from 'Components/RippleButton';
import styles from './style';

/**
 * The action button component.
 */
class ActionButton extends Component {
  static propTypes = {
    ...RippleButton.propTypes,
    onClick: PropTypes.func.isRequired,
    loading: PropTypes.bool,
    noGap: PropTypes.bool,
  };

  static defaultProps = {
    ...RippleButton.defaultProps,
    loading: false,
    type: 'primary',
    flat: true,
    noGap: false,
  };

  static clickDelay = 300;

  /**
   * The click handler
   * @param {Object} event The event object for the click handler
   */
  handleClick = (event) => {
    const { clickDelay } = this.constructor;

    setTimeout(() => {
      this.props.onClick(event);
    }, clickDelay);
  };

  /**
   * Getter for the calculated button props.
   * @returns {Object}
   */
  get buttonProps() {
    const buttonProps = {
      className: this.props.className,
      disabled: this.props.disabled,
      flat: this.props.flat,
      type: this.props.type,
    };

    return buttonProps;
  }

  /**
   * The render function.
   * @returns {JSX}
   */
  render() {
    const containerClass = this.props.noGap ? styles.noGapContainer : styles.container;

    if (this.props.loading) {
      return (
        <div className={styles.containerCircle}>
          <IndicatorCircle />
        </div>
      );
    }

    return (
      <div className={containerClass}>
        <RippleButton {...this.buttonProps} onClick={this.handleClick}>
          {this.props.children}
        </RippleButton>
      </div>
    );
  }
}

export default ActionButton;
