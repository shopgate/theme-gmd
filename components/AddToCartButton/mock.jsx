/**
 * Copyright (c) 2017-present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
/**
 * Mocked AddToCartButton.
 * @type {MockedAddToCartButton}
 */
// eslint-disable-next-line react/prefer-stateless-function
export const MockedAddToCartButton = class extends Component {
  static propTypes = {
    handleAddToCart: PropTypes.func,
  };
  static defaultProps = {
    handleAddToCart: () => {},
  };
  /**
   * Renders mocked button.
   * @return {JSX}
   */
  render() {
    return (
      <button onClick={this.props.handleAddToCart} />
    );
  }
};
