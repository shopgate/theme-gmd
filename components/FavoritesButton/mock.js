/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

const mockedProduct1 = {
  productId: 'foo',
};

const currentProduct = {
  productId: 'bar',
};

/**
 * Mocked state with product only. Favorites not set.
 * @type {Object}
 */
export const mockedStateNotOnList = {
  product: {
    currentProduct,
    productsById: {
      [mockedProduct1.productId]: {
        productData: mockedProduct1,
      },
    },
  },
  favorites: {
    products: {
      ids: [mockedProduct1.productId],
    },
  },
};

/**
 * Mocked state with product only. Favorites not set.
 * @type {Object}
 */
export const mockedStateOnList = {
  product: {
    currentProduct,
    productsById: {
      [mockedProduct1.productId]: {
        productData: mockedProduct1,
      },
      [currentProduct.productId]: {
        productData: currentProduct,
      },
    },
  },
  favorites: {
    products: {
      ids: [mockedProduct1.productId, currentProduct.productId],
    },
  },
};

/**
 * Mocked state without data.
 * @type {{product: {currentProduct: {}, favorites: {}}}}
 */
export const mockedStateEmpty = {
  product: {
    currentProduct: {},
    productsById: {},
  },
  favorites: {
    products: {
      ids: [],
    },
  },
};
