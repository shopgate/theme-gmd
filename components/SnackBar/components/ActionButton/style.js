/**
 *  Copyright (c) 2017 - present, Shopgate, Inc. All rights reserved.
 *
 *  This source code is licensed under the Apache 2.0 license found in the
 *  LICENSE file in the root directory of this source tree.
 *
 */
import { css } from 'glamor';
import colors from 'Styles/colors';

const actionButton = css({
  color: colors.primary,
  flexGrow: 1,
  textTransform: 'uppercase',
  textAlign: 'right',
  padding: '7px 0',
  wordBreak: 'keep-all',
  ':focus': {
    outline: 'none',
  },
}).toString();

export default {
  actionButton,
};
