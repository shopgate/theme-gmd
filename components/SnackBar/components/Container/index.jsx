/**
 *  Copyright (c) 2018, Shopgate, Inc. All rights reserved.
 *
 *  This source code is licensed under the Apache 2.0 license found in the
 *  LICENSE file in the root directory of this source tree.
 *
 */
import React from 'react';
import PropTypes from 'prop-types';
import styles from './style';
/**
 * Container component.
 * @param {Object} props Props.
 * @returns {function}
 */
const Container = props => (
  <div className={styles.container}>
    {props.children}
  </div>
);

Container.prototype.propTypes = {
  children: PropTypes.arrayOf(PropTypes.node),
};
Container.prototype.defaultProps = {
  children: [],
};

export default Container;
