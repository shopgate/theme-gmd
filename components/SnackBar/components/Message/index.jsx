/**
 *  Copyright (c) 2018, Shopgate, Inc. All rights reserved.
 *
 *  This source code is licensed under the Apache 2.0 license found in the
 *  LICENSE file in the root directory of this source tree.
 *
 */
import I18n from '@shopgate/pwa-common/components/I18n';
import React from 'react';
import PropTypes from 'prop-types';
import styles from './style';

/**
 * Message component.
 * @returns {function}
 */
const Message = ({ text }) => (
  <p className={styles.message}>
    <I18n.Text string={text} />
  </p>
);

Message.propTypes = {
  text: PropTypes.string,
};

Message.defaultProps = {
  text: null,
};

export default Message;
