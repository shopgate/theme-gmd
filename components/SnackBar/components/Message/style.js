/**
 *  Copyright (c) 2017 - present, Shopgate, Inc. All rights reserved.
 *
 *  This source code is licensed under the Apache 2.0 license found in the
 *  LICENSE file in the root directory of this source tree.
 *
 */

import { css } from 'glamor';

const message = css({
  color: 'white',
  margin: 0,
  textAlign: 'left',
  maxWidth: '75%',
  paddingTop: 7,
  paddingBottom: 7,
}).toString();

export default {
  message,
};
