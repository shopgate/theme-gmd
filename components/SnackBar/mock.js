/**
 * Copyright (c) 2017 - present, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

/**
 * Mocked toast state
 * @type {{toast: [null,null]}}
 */
export const mockedState = {
  toast: [
    {
      id: 1,
      message: 'Lorem ipsum dolor sit amet.',
      duration: 6000,
    },
  ],
};
