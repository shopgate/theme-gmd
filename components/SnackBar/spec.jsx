/**
 * Copyright (c) 2018, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';
import mockRenderOptions from '@shopgate/pwa-common/helpers/mocks/mockRenderOptions';
import Message from './components/Message';
import Container from './components/Container';
import ActionButton from './components/ActionButton';
import { mockedState } from './mock';

const mockedStore = configureStore();

const createComponent = (state) => {
  /* eslint-disable global-require */
  const SnackBar = require('./index').default;
  /* eslint-enable global-require */
  return mount(
    <Provider store={mockedStore(state)}>
      <SnackBar />
    </Provider>,
    mockRenderOptions
  );
};

describe('<SnackBar />', () => {
  it('should render component without action', () => {
    const wrapper = createComponent(mockedState);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(Message).exists()).toBe(true);
    expect(wrapper.find(Container).exists()).toBe(true);
  });
  it('shuld render component with action button', () => {
    const wrapper = createComponent({
      toast: [
        {
          ...mockedState.toast[0],
          action: 'click me',
        },
      ],
    });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(Message).exists()).toBe(true);
    expect(wrapper.find(Container).exists()).toBe(true);
    expect(wrapper.find(ActionButton).exists()).toBe(true);
  });
});
