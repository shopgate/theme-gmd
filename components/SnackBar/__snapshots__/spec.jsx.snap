// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SnackBar /> should render component without action 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <SnackBar>
    <Connect(Toast)
      actionButton={[Function]}
      className="css-1nw5ecr"
      container={[Function]}
      message={[Function]}
    >
      <Toast
        actionButton={[Function]}
        className="css-1nw5ecr"
        container={[Function]}
        dispatchAction={[Function]}
        message={[Function]}
        onClose={[Function]}
        removeToast={[Function]}
        toast={
          Object {
            "duration": 6000,
            "id": 1,
            "message": "Lorem ipsum dolor sit amet.",
          }
        }
      >
        <div
          aria-live="polite"
          aria-relevant="additions"
        >
          <Drawer
            alwaysActive={false}
            animation={
              Object {
                "duration": null,
                "in": "",
                "out": "",
              }
            }
            className="css-1nw5ecr"
            isOpen={true}
            onClose={[Function]}
            onDidClose={[Function]}
            onOpen={[Function]}
          >
            <div
              className="css-1nw5ecr css-1rg7udn css-1dwe0qx"
              onAnimationEnd={[Function]}
              style={Object {}}
            >
              <Container>
                <div
                  className="css-q9taby"
                >
                  <Message
                    text="Lorem ipsum dolor sit amet."
                  >
                    <p
                      className="css-pnkcbw"
                    >
                      <Translate
                        className=""
                        params={Object {}}
                        string="Lorem ipsum dolor sit amet."
                      >
                        <span
                          className=""
                        >
                          Lorem ipsum dolor sit amet.
                        </span>
                      </Translate>
                    </p>
                  </Message>
                </div>
              </Container>
            </div>
          </Drawer>
        </div>
      </Toast>
    </Connect(Toast)>
  </SnackBar>
</Provider>
`;

exports[`<SnackBar /> shuld render component with action button 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <SnackBar>
    <Connect(Toast)
      actionButton={[Function]}
      className="css-1nw5ecr"
      container={[Function]}
      message={[Function]}
    >
      <Toast
        actionButton={[Function]}
        className="css-1nw5ecr"
        container={[Function]}
        dispatchAction={[Function]}
        message={[Function]}
        onClose={[Function]}
        removeToast={[Function]}
        toast={
          Object {
            "action": "click me",
            "duration": 6000,
            "id": 1,
            "message": "Lorem ipsum dolor sit amet.",
          }
        }
      >
        <div
          aria-live="polite"
          aria-relevant="additions"
        >
          <Drawer
            alwaysActive={false}
            animation={
              Object {
                "duration": null,
                "in": "",
                "out": "",
              }
            }
            className="css-1nw5ecr"
            isOpen={true}
            onClose={[Function]}
            onDidClose={[Function]}
            onOpen={[Function]}
          >
            <div
              className="css-1nw5ecr css-1rg7udn css-1dwe0qx"
              onAnimationEnd={[Function]}
              style={Object {}}
            >
              <Container>
                <div
                  className="css-q9taby"
                >
                  <Message
                    text="Lorem ipsum dolor sit amet."
                  >
                    <p
                      className="css-pnkcbw"
                    >
                      <Translate
                        className=""
                        params={Object {}}
                        string="Lorem ipsum dolor sit amet."
                      >
                        <span
                          className=""
                        >
                          Lorem ipsum dolor sit amet.
                        </span>
                      </Translate>
                    </p>
                  </Message>
                  <ActionButton
                    onClick={[Function]}
                    text="click me"
                  >
                    <button
                      className="css-1tbknzy"
                      onClick={[Function]}
                    >
                      <Translate
                        className=""
                        params={Object {}}
                        string="click me"
                      >
                        <span
                          className=""
                        >
                          click me
                        </span>
                      </Translate>
                    </button>
                  </ActionButton>
                </div>
              </Container>
            </div>
          </Drawer>
        </div>
      </Toast>
    </Connect(Toast)>
  </SnackBar>
</Provider>
`;
